conda_package_streaming/extract.py

```diff
@@ -44,7 +45,14 @@ def checked_members():
                 yield member
 
         try:
-            tar_file.extractall(path=dest_dir, members=checked_members())
+            if HAS_TAR_FILTER:
+                tar_file.extractall(
+                    path=dest_dir,
+                    members=checked_members(),
+                    filter=tarfile.fully_trusted_filter,
```

Comment:
From reading https://docs.python.org/3/library/tarfile.html#tarfile-extraction-filter closely:
> The filter argument to [TarFile.extract()](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.extract) or [extractall()](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.extractall) can be:
>
> - the string 'fully_trusted': Honor all metadata as specified in the archive. Should be used if the user trusts the archive completely, or implements their own complex verification.
>
> - the string 'tar': Honor most tar-specific features (i.e. features of UNIX-like filesystems), but block features that are very likely to be surprising or malicious. See [tar_filter()](https://docs.python.org/3/library/tarfile.html#tarfile.tar_filter) for details.
>
> - the string 'data': Ignore or block most features specific to UNIX-like filesystems. Intended for extracting cross-platform data archives. See [data_filter()](https://docs.python.org/3/library/tarfile.html#tarfile.data_filter) for details.
>
> - None (default): Use [TarFile.extraction_filter](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.extraction_filter).

```suggestion
                    filter="fully_trusted",
```

I'm aware that `_get_filter_function` will check for the callable internally, but that's an implementation detail and we should use the standard API.

Reply:
stringly typed API, yuck.